### ISSUE TITLE
1.1 drops support for Django 1.7?

### ISSUE THREAD

**Issue by robert-kisteleki** [jezdez/django-constance#115]
The doc claims that 1.1 adds support for Django 1.8 and 1.9, while <=1.6 are not supported any more. That's fine, except that 1.7(.10) support also seems to be gone. When clicking on the config in django admin, one gets:

TypeError at /admin/constance/config/
object of type 'Config' has no len()
...
Exception Location: 	.../python2.7/site-packages/django/utils/functional.py in inner, line 225

This is while using pyhon 2.7. The net result is that one has to upgrade constance exactly at the time of a django upgrade, which is suboptimal. Is this really the case?

**Comment by jezdez** [comment]
can you paste a full traceback please?

**Comment by robert-kisteleki** [comment]
It's a 350kb email, not that helpful ;-)

I'll check if I can trim it in a dev environment

**Comment by robert-kisteleki** [comment]
OK, at this point all we know is that it breaks in the django template constance/templates/admin/constance/change_list.html, at line 63

However, it seems to be working on a vanilla django install, so the error is probably in the interaction with some other apps and/or ordering. I'll write more if I find anything.

**Comment by Jterbeest** [comment]
Hi,

We've traced the problem to the template context processor:
`return {"config": constance.config}`

If we replace this to for instance:
`return {"constance_config": constance.config}`

everything works like expected.

**Comment by jezdez** [comment]
The context processor hasn't changed for quite a while so I'm not sure what I can do from here. A full traceback is the only way I can help debug this.

**Comment by Jterbeest** [comment]
Here's the full traceback. I doubt it will be helpful as the error has somehow to do with the value of config as it's passed to the template and the trace only show what parts of Django blow up.
```
Environment:


Request Method: GET
Request URL: https://admin-dev.atlas.ripe.net/admin/constance/config/

Django Version: 1.7.10
Python Version: 2.7.5
Installed Applications:
['polymorphic',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.sites',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'django.contrib.redirects',
 'autocomplete_light',
 'constance',
 'constance.backends.database',
 'django_countries',
 'django_extensions',
 'django_tables2',
 'mathfilters',
 'rest_framework',
 'treebeard',
 'markers',
 'ripe_access',
 'atlas',
 'atlas_common',
 'atlas_shovel',
 'atlas_heartbeat',
 'atlas_task',
 'accounting',
 'advertising',
 'ambassadors',
 'anchor_measurements',
 'anchors',
 'apply',
 'atlas_messages',
 'atlas_tools',
 'atlas_logging',
 'auditing',
 'cartography',
 'community',
 'core',
 'cs',
 'dashboard',
 'docs',
 'eastereggs',
 'emails',
 'geocoding',
 'keys',
 'measurements',
 'legal',
 'page',
 'preregister',
 'promotions',
 'probes',
 'sponsors',
 'statistics',
 'targets',
 'ui',
 'lir_tools',
 'feeds',
 'tasks',
 'domainmon',
 'django.contrib.admin',
 'admin_core',
 'sqlgrants',
 'debug_toolbar']
Installed Middleware:
['django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.contrib.redirects.middleware.RedirectFallbackMiddleware',
 'ripe_access.middleware.RipeAccessAuthenticationMiddleware',
 'ripe_access.middleware.RipeAccessAuthenticationMiddleware',
 'debug_toolbar.middleware.DebugToolbarMiddleware']


Template error:
In template /home/atlas/python/lib/python2.7/site-packages/constance/templates/admin/constance/change_list.html, error at line 63
   object of type 'Config' has no len()
   53 :             {% endif %}


   54 :             <table cellspacing="0" id="result_list">


   55 :                 <thead>


   56 :                 <tr>


   57 :                     <th><div class="text">{% trans "Name" %}</div></th>


   58 :                     <th><div class="text">{% trans "Default" %}</div></th>


   59 :                     <th><div class="text">{% trans "Value" %}</div></th>


   60 :                     <th><div class="text">{% trans "Is modified" %}</div></th>


   61 :                 </tr>


   62 :                 </thead>


   63 :                  {% for item in config %} 


   64 :                 <tr class="{% cycle 'row1' 'row2' %}">


   65 :                     <th>{{ item.name }}


   66 :                         <div class="help">{{ item.help_text|linebreaksbr }}</div>


   67 :                     </th>


   68 :                     <td>


   69 :                         {{ item.default }}


   70 :                     </td>


   71 :                         <td>


   72 :                             {{ item.form_field.errors }}


   73 :                             {{ item.form_field }}


Traceback:
File "/home/atlas/python/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  137.                 response = response.render()
File "/home/atlas/python/lib/python2.7/site-packages/django/template/response.py" in render
  103.             self.content = self.rendered_content
File "/home/atlas/python/lib/python2.7/site-packages/django/template/response.py" in rendered_content
  80.         content = template.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/base.py" in render
  148.             return self._render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
  88.     return self.nodelist.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  126.         return compiled_parent._render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
  88.     return self.nodelist.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  126.         return compiled_parent._render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
  88.     return self.nodelist.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  65.                 result = block.nodelist.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/atlas/python/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  162.             len_values = len(values)
File "/home/atlas/python/lib/python2.7/site-packages/django/utils/functional.py" in inner
  225.         return func(self._wrapped, *args)

Exception Type: TypeError at /admin/constance/config/
Exception Value: object of type 'Config' has no len()
```

**Comment by jezdez** [comment]
Aha, thanks, so it happens while rendering the list of config values. I have an idea which change triggered that.

**Comment by jezdez** [comment]
1.1.1 is out with the fix. Thanks!

**Comment by Jterbeest** [comment]
Thanks, we'll test it on Monday :)

**Comment by jezdez** [comment]
Sure, thanks for using constance ;)

**Comment by adamrt** [comment]
Just a heads up, on Django 1.8.1 I had to revert to version 1.1 instead of 1.1.1. I was getting this error, but I goes away on 1.1.

Added context processor
Added installed_apps
Added constance_config
Added constance_backend string for redis

**Comment by jezdez** [comment]
@adamrt Huh, what error?

**Comment by adamrt** [comment]
Hey sorry for not being more specific. I was getting the error for this issue report that has been closed.

`object of type 'Config' has no len()` - in admin on change_list view

I found this closed issue report and reverted to 1.1 from 1.1.1 since I was already on 1.1.1 and the issue was fixed.

I am running Grappelli though which causes an error with boolean fields being required as well, so that could be my root issue. I am going to do some more digging though to see if I could provide more information.
